move-vm-backend/src/warehouse.rs

```diff
@@ -62,45 +74,74 @@ impl AccountData {
 }
 
 /// Move VM storage implementation for Substrate storage.
-pub(crate) struct Warehouse<S: Storage> {
+pub(crate) struct Warehouse<S: Storage, Api: SubstrateAPI> {
     /// Substrate storage implementing the Storage trait
     storage: S,
+    substrate_api: Api,
 }
 
-impl<S: Storage> Warehouse<S> {
-    pub(crate) fn new(storage: S) -> Warehouse<S> {
-        Self { storage }
+impl<S: Storage, Api: SubstrateAPI> Warehouse<S, Api> {
+    pub(crate) fn new(storage: S, substrate_api: Api) -> Warehouse<S, Api> {
+        Self {
+            storage,
+            substrate_api,
+        }
     }
 
     pub(crate) fn apply_changes(&self, changeset: ChangeSet) -> Result<()> {
         for (account, changeset) in changeset.into_inner() {
             let key = account.as_slice();
-            let mut account = match self.storage.get(key) {
+            let mut store_account = match self.storage.get(key) {
                 Some(value) => bcs::from_bytes(&value).map_err(Error::msg)?,
                 _ => AccountData::default(),
             };
 
             let (modules, resources) = changeset.into_inner();
-            AccountData::apply_changes(&mut account.modules, modules)?;
-            AccountData::apply_changes(&mut account.resources, resources)?;
+            let mut unprocessed_resources = vec![];
+            AccountData::apply_changes(&mut store_account.modules, modules)?;
+            // process Deposit
+            for (tag, res) in resources
+                .into_iter()
+                .filter(|r| r.0.eq(&DEPOSIT_TEMPLATE))
```

Comment:
Hmhm, this might drop all other resources, and then we might always end up with an empty `unprocessed_resources` vector.

We do have a check on line 112 which acts like a similar filter. We should check if all tests still pass with this change.

Reply:
Okay, I just ran all the tests and it seems like `get_resource` test fails by default with this PR, but when I remove this line, then it works
```
.filter(|r| r.0.eq(&DEPOSIT_TEMPLATE))
```

Reply:
great catch on this one - initial implementation had no `unprocessed_resources` hence required filtering, now I've removed entire into_iter.